src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -143,6 +144,14 @@ uct_cuda_ipc_mem_add_reg(void *addr, uct_cuda_ipc_memh_t *memh,
         goto legacy_path;
     }
 
+    status = UCT_CUDADRV_FUNC_LOG_ERR(cuPointerGetAttribute(
```

Comment:
1. do we really have to check this, or cuMemExportToShareableHandle would fail anyway?
2. can we unite it with checking CU_POINTER_ATTRIBUTE_IS_LEGACY_CUDA_IPC_CAPABLE to check several attributes at once?
3. Is the goal of the PR is to move the failure from memory registration to rkey unpack? Which test would fail without this PR?

Reply:
1. It would fail, but it may also fail due to other reasons. 
3. Sending  VMM memory allocated without fabric capability will fail (registration failure will lead to MPI_Send abortion)

Reply:
@yosefe 
1. Right now we're checking to avoid more checks in case memory isn't fabric capable. Export can still fail and we want to go through same handling
2. We can combine. I'll make the change. 
3. yes this is the goal of the PR.

Reply:
1. ok